src/Form/ContentpoolChannelForm.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace Drupal\contentpool_client\Form;
+
+use Drupal\contentpool_client\RemotePullManagerInterface;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\relaxed\Entity\RemoteInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class ContentpoolChannelForm.
+ */
+class ContentpoolChannelForm extends FormBase {
```

Comment:
It should be generic to whatever term reference field is on the article. Thus, I'd suggest we refactor this into a "ContentpoolSubscriptionFilterForm"